pkg/relayer/interface.go

```diff
@@ -15,17 +15,17 @@ type RelayerSessionsManager interface {
 	// SessionsToClaim returns an observable that notifies of sessions ready to be claimed.
 	SessionsToClaim() observable.Observable[SessionTree]
 
-	// EnsureSessionTree returns the SMST (Sparse Merkle State Tree) for a given session.
+	// EnsureSessionTree returns the SMST (Sparse Merkle State Tree) for a given session header.
 	// It is used to retrieve the SMST and update it when a Relay has been successfully served.
-	// If the session is seen for the first time, it creates a new SMST for it before returning it.
+	// If the session header is seen for the first time, it creates a new SMST for it before returning it.
```

Comment:
It's still the session that determines the 1-1 mapping. The header is just what carries it.

```suggestion
	// If the session is seen for the first time, it creates a new SMST for it before returning it.
```